### PR TITLE
Add optional pandas dep and ability to fetch a response field as a DataFrame

### PR DESCRIPTION
**Pull request by mschroering**
```python
>>> from phc import Session
>>> session = Session(account="lifeomic")
>>> from phc.services import Accounts
>>> accounts = Accounts(session)
>>> res = accounts.get_list()
>>> df = res.get_dataFrame("accounts")
>>> df = res.get_dataFrame("accounts", mapFunc=lambda r: r.get("name"))
```

### CODE REVIEW

File: phc/api_response.py [hunk 1]
```diff
@@ -83,6 +91,35 @@ def get(self, key: str, default=None):
 
         return self.data.get(key, default)
 
+    def get_dataFrame(self, key: str, mapFunc: Callable[[Any], Any] = None):
```
Comment (mschroering) [inline phc/api_response.py]: Open to suggestions on this name.  Was struggling with what to call it.

Comment (mjtieman) [reply phc/api_response.py]: Maybe `get_as_dataframe`. Do you have an example to show expected usage?

Comment (mschroering) [reply phc/api_response.py]:
```python
>>> from phc import Session
>>> session = Session(account="lifeomic")
>>> from phc.services import Accounts
>>> accounts = Accounts(session)
>>> res = accounts.get_list()
>>> df = res.get_dataFrame("accounts")
>>> df = res.get_dataFrame("accounts", mapFunc=lambda r: r.get("name"))
```

Comment (mschroering) [reply phc/api_response.py]: @mjtieman updated to:
```python
>>> from phc import Session
>>> session = Session(account="lifeomic")
>>> from phc.services import Accounts
>>> accounts = Accounts(session)
>>> res = accounts.get_list()
>>> df = res.get_as_dataFrame("accounts")
>>> df = res.get_as_dataFrame("accounts", mapFunc=lambda r: r.get("name"))
```

Comment (mjtieman) [reply phc/api_response.py]: Super minor: I would rather not mix camel case and underscores, mind renaming to `get_as_dataframe`?

Comment (mschroering) [reply phc/api_response.py]: Good suggestion. Updated the code.

**Pull request merged**